priam/src/main/java/com/netflix/priam/identity/token/DeadTokenRetriever.java

```diff
@@ -107,45 +88,84 @@ public PriamInstance get() throws Exception {
+            // test same zone and is it alive.
+            if (!priamInstance.getRac().equals(instanceInfo.getRac())
+                    || asgInstances.contains(priamInstance.getInstanceId())
+                    || super.isInstanceDummy(priamInstance)) continue;
+            // TODO: If instance is in SHUTTING_DOWN mode, it might not show up in asg instances (if
+            // cloud control plane is having issues), thus, we should not try to replace the
+            // instance as it will lead to "Cannot replace a live node" issue.
+
+            logger.info("Found dead instance: {}", priamInstance.toString());
+
             PriamInstance markAsDead =
                     factory.create(
-                            dead.getApp() + "-dead",
-                            dead.getId(),
-                            dead.getInstanceId(),
-                            dead.getHostName(),
-                            dead.getHostIP(),
-                            dead.getRac(),
-                            dead.getVolumes(),
-                            dead.getToken());
+                            priamInstance.getApp() + "-dead",
+                            priamInstance.getId(),
+                            priamInstance.getInstanceId(),
+                            priamInstance.getHostName(),
+                            priamInstance.getHostIP(),
+                            priamInstance.getRac(),
+                            priamInstance.getVolumes(),
+                            priamInstance.getToken());
             // remove it as we marked it down...
-            factory.delete(dead);
+            factory.delete(priamInstance);
 
             // find the replaced IP
-            this.replacedIp = findReplaceIp(allIds, markAsDead.getToken(), markAsDead.getDC());
-            if (this.replacedIp == null) this.replacedIp = markAsDead.getHostIP();
+            this.replacedIp =
+                    findReplaceIp(
+                            allInstancesWithinCluster,
+                            priamInstance.getToken(),
+                            priamInstance.getDC());
+            // Lets not replace the instance if gossip info is not merging!!
+            if (replacedIp == null) return null;
 
-            String payLoad = markAsDead.getToken();
             logger.info(
-                    "Trying to grab slot {} with availability zone {}",
-                    markAsDead.getId(),
-                    markAsDead.getRac());
-            return factory.create(
-                    config.getAppName(),
-                    markAsDead.getId(),
-                    instanceInfo.getInstanceId(),
-                    instanceInfo.getHostname(),
-                    instanceInfo.getHostIP(),
-                    instanceInfo.getRac(),
-                    markAsDead.getVolumes(),
-                    payLoad);
+                    "Will try to replace token: {} with replacedIp (from gossip info): {} instead of ip from DB: {}",

```

Comment:
`from DB` --> `from turtle`?

Reply:
turtle is Netflix token DB (not common). Token DB, might be more appropriate.